core/lib/zksync_core/src/api_server/web3/namespaces/snapshots.rs

```diff
@@ -0,0 +1,89 @@
+use crate::api_server::web3::backend_jsonrpc::error::internal_error;
+use crate::api_server::web3::state::RpcState;
+use crate::l1_gas_price::L1GasPriceProvider;
+use zksync_types::snapshots::{AllSnapshots, SnapshotHeader, SnapshotStorageLogsChunkMetadata};
+use zksync_types::L1BatchNumber;
+use zksync_web3_decl::error::Web3Error;
+
+#[derive(Debug)]
+pub struct SnapshotsNamespace<G> {
+    pub state: RpcState<G>,
+}
+
+impl<G> Clone for SnapshotsNamespace<G> {
+    fn clone(&self) -> Self {
+        Self {
+            state: self.state.clone(),
+        }
+    }
+}
+impl<G: L1GasPriceProvider> SnapshotsNamespace<G> {
+    pub fn new(state: RpcState<G>) -> Self {
+        Self { state }
+    }
+    pub async fn get_all_snapshots_impl(&self) -> Result<AllSnapshots, Web3Error> {
+        let method_name = "get_all_snapshots";
+        let mut storage_processor = self
```

Comment:
Nit: We probably want to measure method latency as with other Web3 methods using `API_METRICS`.

Reply:
fixed